downloader/artifact_downloader.go

```diff
@@ -117,10 +119,10 @@ func (ad *ConcurrentArtifactDownloader) downloadFile(targetDir, fileName, downlo
 
 	ctx, cancel := context.WithTimeout(context.Background(), ad.Timeout)
 
-	downloader := got.New()
-	downloader.Client = retry.NewHTTPClient().StandardClient()
+	downloader := got.NewWithContext(ctx)
+	downloader.Client = ad.createClient().StandardClient()
 
-	err := downloader.Do(got.NewDownload(ctx, downloadURL, fileFullPath))
```

Comment:
Discovering why this code piece was not using the retry even if it was set was fun.

Here the retryable http client was set on the downloader but it was not used during execution because the `got.NewDownload` set a default http client for the download and the downloader's client will be ignored. 

So if we want to have retry then we need to initialise the downloader with a context and set the client. Then we have to use the `downloader.Download` function which will use the downloader's http client. 😄